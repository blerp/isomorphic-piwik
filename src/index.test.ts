import { Piwik } from "./index";

describe("isomorphic-piwik", () => {
  let tracker: Piwik;

  beforeAll(() => {
    tracker = new Piwik("localhost/piwik.php", 1);
  });

  beforeEach(() => {
    fetch.resetMocks();
  });

  it("does a basic request", async () => {
    await tracker.report({});
    expect(fetch.mock.calls).toMatchSnapshot();
  });
});
