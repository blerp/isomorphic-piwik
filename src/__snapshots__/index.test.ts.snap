// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isomorphic-piwik posts a basic report with an empty object 1`] = `
Array [
  Array [
    "localhost/piwik.php?apiv=1&idSite=1&rec=1&url=%2F",
    Object {
      "method": "post",
    },
  ],
]
`;

exports[`isomorphic-piwik posts a basic report without an empty object 1`] = `
Array [
  Array [
    "localhost/piwik.php?apiv=1&idSite=1&rec=1&url=%2F",
    Object {
      "method": "post",
    },
  ],
]
`;
